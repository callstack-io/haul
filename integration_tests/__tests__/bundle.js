/**
 * Copyright 2017-present, Callstack.
 * All rights reserved.
 *
 * @flow
 */

const { runHaulSync } = require('../runHaul');
const path = require('path');
const fs = require('fs');
const { cleanup, run, replaceTestPath } = require('../utils');

const TEST_PROJECT_DIR = path.resolve(
  __dirname,
  '../fixtures/react-native-generated-project',
);

beforeAll(() => {
  run('yarn', TEST_PROJECT_DIR);
});

beforeEach(() => cleanup(path.resolve(TEST_PROJECT_DIR, 'dist')));

test('bundle ios project', () => {
  bundleForPlatform('ios');
});

test('bundle android project', () => {
  bundleForPlatform('android');
});

function bundleForPlatform(platform) {
  const bundlePath = path.resolve(
    TEST_PROJECT_DIR,
    'dist',
    `index.${platform}.bundle`,
  );
  const { stdout } = runHaulSync(TEST_PROJECT_DIR, [
    'bundle',
    '--platform',
    platform,
  ]);
  expect(replaceTestPath(stdout.toString())).toMatchSnapshot();
  expect(fs.existsSync(bundlePath));
}
