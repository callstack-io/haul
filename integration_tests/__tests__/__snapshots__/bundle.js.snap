// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bundle android project 1`] = `
"
 INFO  Haul is now bundling your React Native app in production mode.

Starting from:
<<REPLACED>>/fixtures/react-native-generated-project/index.android.js 

[2J[0f DONE  Built successfully!
<<REPLACED>>/fixtures/react-native-generated-project/dist
Bundle location: /Users/thymikee/Projects/callstack-io/haul/integration_tests/fixtures/react-native-generated-project/dist/index.android.bundle
"
`;

exports[`bundle ios project 1`] = `
"
 INFO  Haul is now bundling your React Native app in production mode.

Starting from:
<<REPLACED>>/fixtures/react-native-generated-project/index.ios.js 

[2J[0f DONE  Built successfully!
<<REPLACED>>/fixtures/react-native-generated-project/dist
Bundle location: /Users/thymikee/Projects/callstack-io/haul/integration_tests/fixtures/react-native-generated-project/dist/index.ios.bundle
"
`;
